src/seedu/addressbook/data/AddressBook.java

```diff
@@ -128,4 +140,24 @@ public boolean equals(Object other) {
                         && this.allPersons.equals(((AddressBook) other).allPersons)
                         && this.allTags.equals(((AddressBook) other).allTags));
     }
+
+    /**
+     * Constructs a comparator that compares two Strings in reverse lexicographical order.
+     */
+
+    public class ReverseAlphabeticalComparator implements Comparator<Person> {
+
+        public int compare(Person p1, Person p2){
+            int result = p1.getName().toString().compareTo(p2.getName().toString());
```

Comment:
I think it may be better if we used the compareTo function instead, but invert the output. 

For example, suppose I have two names (in string), p1 and p2 
Might be easier and shorter to just return -1 * p1.compareTo(p2)

Reply:
yes agreed, it will be easier to implement and understand. Because in this case you only have one attribute you need to compare against, so easier to use extend `Comparable` and use `compareTo` function.